### PR TITLE
Feature cxx

### PR DESCRIPTION
**Pull request by Jegp**
* Re-introduced C++ as a fix for #184 
* Added cross-platform builds for Mac and Windows

### CODE REVIEW

File: .github/workflows/cpppackage.yml [hunk 1]
```diff
@@ -0,0 +1,51 @@
+name: C++ package
+
+on: [push, pull_request]
+
+env:
+  # Customize the CMake build type here (Release, Debug, RelWithDebInfo, etc.)
+  BUILD_TYPE: Debug
```
Comment (cpehle) [inline .github/workflows/cpppackage.yml:7]: Shouldn't this be Release by default?

File: norse/torch/functional/test/test_lif.py [hunk 2]
```diff
@@ -2,27 +2,28 @@
...
 
 import norse
-import norse.torch.functional.lif as lif_module
 from norse.torch.functional.lif import (
     LIFState,
     LIFFeedForwardState,
     LIFParameters,
     LIFParametersJIT,
     lif_step,
+    # lif_step_integral,
```
Comment (cpehle) [inline norse/torch/functional/test/test_lif.py:11]: comment?

**Review by cpehle — APPROVED** [review]
Great Job! Found only minor nitpicks.

File: .github/workflows/cpppackage.yml (hunk 1 above)
Comment (Jegp) [reply .github/workflows/cpppackage.yml:7]: I've set tests to build when build type is `Debug`, so for the workflow this is correct.

File: norse/torch/functional/test/test_lif.py (hunk 2 above)
Comment (Jegp) [reply norse/torch/functional/test/test_lif.py]: Good catch! I added the module, including some tests.

**Comment by Jegp** [comment]
Thanks for the review @cpehle!

**Pull request merged**